andino_description/urdf/include/andino_caster_macro.urdf.xacro

```diff
@@ -115,7 +115,8 @@
       </joint>
     </xacro:if>
     <xacro:unless value="${use_fixed}">
-      <joint name="caster_rotation_joint" type="continuous">
+      <!--TODO(olmerg) initial solution problem with ros control  -->
+      <joint name="caster_rotation_joint" type="fixed">
```

Comment:
Not seeing why this change. This section of the if-unless clause is to have continuous joint.

In case you need to use the fixed one in your simulation pass a different argument to the xacro.

Reply:
The problem is that the only solution that I found for ros control works  is with the caster_rotation_joint fixed and the caster_rotation_wheel continuous. I can isolate the error in one file is  the urdf is modified inside the spawn_launch, What do you think?

Reply:
Have you tried modelling the caster wheel collision geometry as a sphere instead of a cillinder? That would mean it will have a single point of contact with the floor, instead of a line of points.

Reply:
Chiming in since I added the option to select the joint's type.
I understand different physics engines may behave differently and might require little tweaks as modeling the caster as a non-friction ball or a real-like implementation with joints.
What I don't understand is why that is a limitation for `ros_control`. You would never want the caster wheel to be controlled/sensed through `ros_control`, right? So why this is a problem?
I feel we're all asking and answering about different things, so knowing what the real problem is might be the best approach to discuss upon.

Reply:
> The problem is that the only solution that I found for ros control works is with the caster_rotation_joint fixed and the caster_rotation_wheel continuous. I can isolate the error in one file is the urdf is modified inside the spawn_launch, What do you think?

My comment is focused on trying to not mess up the logic in the urdf description. With that change, the `use_fixed` arg lacks of consistency. The urdf description can be as flexible as is required to deal with simulator's issues while keeping the integrity of the description.

Reply:
I understand your approach , but the problem that I want to solve is that the robot do not move in straight-line with ros_control activated. Th only way that I have found is to emulate a sphere wheel with wheel friction zero ( caster_rotation_joint fixed and the caster_rotation_wheel continuous).

Reply:
You don't need the caster to be attached to a continuous joint to define the wheel friction to zero. Keep it fixed (as it was before, both joints fixed) and set something along the lines of (not tested and written from the top of my head):
```
<gazebo reference="caster_wheel">
  <mu1>0.0</mu1>
  <mu2>0.0</mu2>
</gazebo>
```

Isn't that enough?

Reply:
this solution did not works I do not understand why, I have tested different combinations in the $\mu$ of the caster wheel and the motor wheels.

Reply:
If you remove the friction from all the wheels, the robot wouldn't move at all.
Let me check on my own if I can make it work. Any suggestion on how to quickly bring it up and test it?

Reply:
following the readme should works